lantern/oop/homework/homework.py

```diff
@@ -30,249 +63,308 @@ class Cat:
       if it runs more than 200(not including) than _reduce_saturation_level with value 50
 
       return text like this: f"Your cat ran {ran_km} kilometers"
-
-    * Implement get_saturation_level and return saturation_level
-      if saturation_level eq 0 return text like this: "Your cat is dead :("
-
-    * Implement get_average_speed and return average_speed
-
     """
 
-    def __init__(self, age):
-        pass
-
-    def eat(self, product):
-        pass
-
-    def _reduce_saturation_level(self, value):
-        pass
-
-    def _increase_saturation_level(self, value):
-        pass
-
-    def _set_average_speed(self):
-        pass
-
     def run(self, hours):
-        pass
+        ran_km = hours * self.average_speed
+        self._reduce_saturation_level({ran_km > 200: 50, ran_km <= 200: 25, ran_km <= 100: 15,
+                                       ran_km <= 50: 5, ran_km <= 25: 2}.get(True))
+        return {f"Your cat ran {ran_km} kilometers"}
+
+    """
+    * Implement get_saturation_level and return saturation_level
+      if saturation_level eq 0 return text like this: "Your cat is died :("
+    """
 
     def get_saturation_level(self):
-        pass
+        return self.saturation_level if self.saturation_level > 0 else "Your cat is died :("
```

Comment:
return always the same type
if you wanna notify user that cat was died, just raise an exception

Reply:
@dimigor If I raise an exception the test will failed